tests/foreman/cli/test_discoveryrule.py

```diff
@@ -638,90 +483,27 @@ def test_positive_create_rule_with_non_admin_user(
         )
         assert rule.name == rule_name
 
-    @pytest.mark.tier2
-    def test_positive_delete_rule_with_non_admin_user(
-        self, class_org, class_location, class_user_manager, class_hostgroup, class_user_password
-    ):
-        """Delete rule with non-admin user by associating discovery_manager role
-
-        :id: 87ab969b-7d92-478d-a5c0-1c0d50e9bdd6
-
-        :expectedresults: Rule should be deleted successfully.
-
-        :CaseLevel: Integration
-        """
-        rule_name = gen_string('alpha')
-        rule = Box(
-            DiscoveryRule.with_user(class_user_manager.login, class_user_password).create(
-                {
-                    'name': rule_name,
-                    'search': 'cpu_count = 5',
-                    'organizations': class_org.name,
-                    'locations': class_location.name,
-                    'hostgroup-id': class_hostgroup.id,
-                }
-            )
-        )
-        rule = Box(
-            DiscoveryRule.with_user(class_user_manager.login, class_user_password).info(
-                {'id': rule.id}
-            )
-        )
-
-        DiscoveryRule.with_user(class_user_manager.login, class_user_password).delete(
-            {'id': rule.id}
-        )
+        target_sat.cli.DiscoveryRule.with_user(
+            class_user_manager.login, class_user_password
+        ).delete({'id': rule.id})
         with pytest.raises(CLIReturnCodeError):
-            DiscoveryRule.info({'id': rule.id})
-
-    @pytest.mark.tier2
-    def test_positive_view_existing_rule_with_non_admin_user(
-        self, class_org, class_location, class_user_password, class_user_reader, class_hostgroup
-    ):
-        """Existing rule should be viewed to non-admin user by associating
-        discovery_reader role.
-
-        :id: 7b1d90b9-fc2d-4ccb-93d3-605c2da876f7
-
-        :Steps:
-
-            1. create a rule with admin user
-            2. create a non-admin user and assign 'Discovery Reader' role
-            3. Login with non-admin user
-
-        :expectedresults: Rule should be visible to non-admin user.
-
-        :CaseLevel: Integration
-        """
-        rule_name = gen_string('alpha')
-        rule = Box(
-            make_discoveryrule(
-                {
-                    'name': rule_name,
-                    'enabled': 'false',
-                    'search': "last_report = Today",
-                    'organizations': class_org.name,
-                    'locations': class_location.name,
-                    'hostgroup-id': class_hostgroup.id,
-                }
-            )
-        )
-        rule = Box(
-            DiscoveryRule.with_user(class_user_reader.login, class_user_password).info(
-                {'id': rule.id}
-            )
-        )
-        assert rule.name == rule_name
+            target_sat.cli.DiscoveryRule.info({'id': rule.id})
 
     @pytest.mark.tier2
     def test_negative_delete_rule_with_non_admin_user(
```

Comment:
We could test negative update for this user too, when user doesn't have a proper permissions, wdyt?

Reply:
That will be a new test case in which check more such things view/create i will try to come up with it in my test case adding PR's